### PR TITLE
wire PrintFlags through rollout commands

### PR DESCRIPTION
**Pull request by juanvallejo**
Binds PrintFlags to rollout commands.
Adds tests ensuring --template printing is supported by rollout cmds.

Fixes https://github.com/kubernetes/kubernetes/issues/65778

**Release note**:
```release-note
NONE
```

cc @soltysh @deads2k

### CODE REVIEW

**Comment by neolit123** [comment]
not sure if `pull-kubernetes-integration` is flaking here or if there is an error with `template-output.sh`.
/test pull-kubernetes-integration

/cc @kubernetes/sig-cli-pr-reviews

File: pkg/kubectl/cmd/rollout/rollout.go [hunk 1]
```diff
@@ -56,10 +56,10 @@ func NewCmdRollout(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobr
 		Run:     cmdutil.DefaultSubCommandRun(streams.Out),
 	}
 	// subcommands
-	cmd.AddCommand(NewCmdRolloutHistory(f, streams.Out))
+	cmd.AddCommand(NewCmdRolloutHistory(f, streams))
```
Comment (soltysh) [inline pkg/kubectl/cmd/rollout/rollout.go]: That's unfortunate, do we have more like these?

File: pkg/kubectl/cmd/rollout/rollout_history.go [hunk 2]
```diff
@@ -54,65 +81,93 @@ func NewCmdRolloutHistory(f cmdutil.Factory, out io.Writer) *cobra.Command {
...
 }
 
-func RunHistory(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, args []string, options *resource.FilenameOptions) error {
-	if len(args) == 0 && cmdutil.IsFilenameSliceEmpty(options.Filenames) {
-		return cmdutil.UsageErrorf(cmd, "Required resource not specified.")
+func (o *RolloutHistoryOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.Resources = args
+
+	var err error
+	if o.Namespace, o.NamespaceRequired, err = f.ToRawKubeConfigLoader().Namespace(); err != nil {
```
Comment (soltysh) [inline pkg/kubectl/cmd/rollout/rollout_history.go]: s/`NamepaceRequired`/`ExplicitNamespace`

File: pkg/kubectl/cmd/rollout/rollout_resume.go [hunk 3]
```diff
@@ -92,17 +100,21 @@ func NewCmdRolloutResume(f cmdutil.Factory, streams genericclioptions.IOStreams)
...
 		return cmdutil.UsageErrorf(cmd, "%s", cmd.Use)
 	}
 
+	o.Resources = args
+
 	o.Resumer = polymorphichelpers.ObjectResumerFn
 
-	cmdNamespace, enforceNamespace, err := f.ToRawKubeConfigLoader().Namespace()
+	var err error
+	o.Namespace, o.NamespaceRequired, err = f.ToRawKubeConfigLoader().Namespace()
```
Comment (soltysh) [inline pkg/kubectl/cmd/rollout/rollout_resume.go]: `ExplicitNamespace`

File: pkg/kubectl/cmd/rollout/rollout_undo.go [hunk 4]
```diff
@@ -98,19 +102,20 @@ func NewCmdRolloutUndo(f cmdutil.Factory, out io.Writer) *cobra.Command {
...
 	}
 
-	o.Out = out
+	o.Resources = args
 	o.DryRun = cmdutil.GetDryRunFlag(cmd)
 
-	cmdNamespace, enforceNamespace, err := f.ToRawKubeConfigLoader().Namespace()
-	if err != nil {
+	var err error
+	if o.Namespace, o.NamespaceRequired, err = f.ToRawKubeConfigLoader().Namespace(); err != nil {
```
Comment (soltysh) [inline pkg/kubectl/cmd/rollout/rollout_undo.go]: `ExplicitNamespace`

**Review by soltysh — CHANGES_REQUESTED** [review]
A few nits...

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by juanvallejo** [comment]
cc @deads2k for re-lgtm

**Comment by deads2k** [comment]
just additional tests

/lgtm

**Pull request merged**